application/plugin_base/item_list.py

```diff
@@ -1655,11 +1673,34 @@ def get_provider(self, path=None):
 		if path is None:
 			return self._current_provider
 
+		# create a local provider if no provider exists yet as fallback in case we are opening an archive
+		if len(self._providers) == 0:
+
+			Provider = self._parent.get_provider_by_protocol('file')
+			result = Provider(self, user.home)
+
+			# cache provider for later use
+			root_path = result.get_root_path(user.home)
+			self._providers[root_path] = result
+			self._current_provider = result
+
 		# check if there is a provider for specified path
 		if path in self._providers:
 			result = self._providers[path]
 
 		else:
+
+			# assuming a path like /home/user/some.zip/subdir1/subdir2
+			# creates intermediate archive provider if necessary
+			p = path
+			while p not in self._providers and p != '/' and '://' not in path:
```

Comment:
Creating `LocalProvider` by default is not good. If path is say on FTP, default provider would need to be one for FTP, not `LocalProvider`. Providers should be created as they are needed. Now you point out a problem which I overlooked initially, when restoring full path and not entering manually level by level.

I asked what that check does because it wasn't initially clear to me but now I see that you are actually restricting archive providers only to local file system and excluding all URI formed paths. This is not good and we should not decide for user when they want to access archive.